.gitignore

```diff
@@ -2,6 +2,7 @@ Cargo.lock
 target/
 rls/
 .vscode/
+.idea/
```

Comment:
I'm not sure why it should be added. In general, I don't like the entries in that file, so let's not add any extra.

For such things you have a global gitignore file, so please remove it from here.

Reply:
Yeah I agree with that. The existing .vscode files are already kinda useless. People should maintain their global gitignores instead of having to add this to every single project.

Reply:
Yeah, I kinda forgot I had added that when making the PR. Only even did so because every other project I've ever contributed to had it, and `.vscode/` was already there. Also sometimes for for personal projects I don't ignore `.idea/`, but that's of questionable value anyway.